txstratum/protocol.py

```diff
@@ -151,7 +154,13 @@ def method_subscribe(self, params: Any, msgid: JSONRPCId) -> None:
         if params and 'address' in params and params['address'] is not None:
             try:
                 address = params['address']
-                self.miner_address = decode_address(address)
+                address_bytes = decode_address(address)
```

Comment:
Should we accept this only for testnet? Or maybe add a parameter to allow it, I don't know. As you said in the description, it makes sense in the context but no sense for code context.

And where do we raise `InvalidAddress` error? Inside `decode_address` method? Do we validate the network byte there?

Reply:
Honestly I think we can safely accept (logging a warning) a wrong address on the testnet for the purposes of mining blocks, tx-mining-service has the least chance to find a block on the mainnet (least chance is maybe putting it lightly, it has absolutely no chance).

Reply:
`decode_address` will raise `InvalidAddress` when it has the wrong length or when the checksum doesn't match, but it will not validate the network byte, this is only done on `create_output_script` much later.

Reply:
Yeah, I agree with you